### ISSUE TITLE
Unable to manage 404 responses by CircleCI

### ISSUE THREAD

**Issue by negipo** [Kesin11/CIAnalyzer#1120]
Our project uses CIAnalyzer v6.0.2 for CircleCI, but recently we realized that CIAnalyzer is unable to manage 404 responses.

```
ERROR   [_CircleciRunnerV2:_CircleciClientV2]{
  message: 'Request failed with status code 404',
  request: {
    method: 'GET',
    host: 'circleci.com',
    path: '/api/v2/project/gh/[project]/job/127418'
  },
  response: {
    status: 404,
    statusText: 'Not Found',
    baseUrl: 'https://circleci.com/api',
    url: 'v2/project/gh/[project]/job/127418',
    params: undefined
  }
}
```

[whole log](https://gist.github.com/negipo/475fbe9cf71b1cbf63be2b4649146d60)

We have taken the solution described [here](https://github.com/Kesin11/CIAnalyzer/issues/518#issuecomment-1091912017), but since it is occurring frequently, I am reporting here.
I think the issue is similar to https://github.com/Kesin11/CIAnalyzer/issues/518

**Comment by Kesin11** [comment]
@negipo 
Thank you for reporting this issue. I would like to know more about the situation you faced, any idea about the cause?
For example, is there a CircleCI project setup that might be related?

**Comment by Kesin11** [comment]
@negipo 
I looked into the case where the response to `/api/v2/project/gh/[project]/job/{job_number}` was 404, most likely due to the return value filtering of fetchWorkflowJobs() not working.
https://github.com/Kesin11/CIAnalyzer/blob/89ec4a145fe16af0344612ad4d0291210d09c268/src/client/circleci_client_v2.ts#L308-L318

I would like to determine why this filtering is not working, so please let me know the [workflow jobs API](https://github.com/Kesin11/CIAnalyzer/blob/89ec4a145fe16af0344612ad4d0291210d09c268/src/client/circleci_client_v2.ts#L336-L340) response to the workflow that is causing the problem for your project.

For example, The JSON response of [this page]( https://app.circleci.com/pipelines/github/Kesin11/CIAnalyzer/3797/workflows/ca0d082d-6362-4d2f-b3f2-e71c01c02406) can be fetched using curl.

```
$ curl -sS -u "$CIRCLECI_TOKEN:" https://circleci.com/api/v2/workflow/ca0d082d-6362-4d2f-b3f2-e71c01c02406/job
{
  "next_page_token" : null,
  "items" : [ {
    "job_number" : 11225,
    "started_at" : "2024-03-08T15:30:57Z",
    "name" : "lint",
    "project_slug" : "gh/Kesin11/CIAnalyzer",
    "type" : "build",
    "status" : "canceled",
    "id" : "68692c9f-b336-4564-930a-06f558374297",
    "dependencies" : [ ]
  }, {
    "started_at" : null,
    "name" : "build_and_test",
    "project_slug" : "gh/Kesin11/CIAnalyzer",
    "type" : "build",
    "status" : "canceled",
    "id" : "ab9ac829-f74b-45bf-b123-3147f9516172",
    "dependencies" : [ "68692c9f-b336-4564-930a-06f558374297" ]
  } ]
}
```

ref: https://circleci.com/docs/api/v2/index.html#operation/listWorkflowJobs

**Comment by negipo** [comment]
@Kesin11 Thank you for quick reply.
I have checked the API response and it was followings:

```
curl -sS -u "$CIRCLECI_TOKEN:" https://circleci.com/api/v2/workflow/8506fdb3-dfd7-4b47-92f0-21cf5bcadc4a/job

{
  "next_page_token" : null,
  "items" : [ {
    "dependencies" : [ ],
    "job_number" : 127418,
    "id" : "a88020fd-c5c3-48ac-bc74-25a88593c08b",
    "started_at" : "2024-03-06T09:33:14Z",
    "name" : "feature_branch_pr_build_runner_start",
    "approved_by" : "f805acbb-1026-4caf-b0a0-1e9190042c72",
    "project_slug" : "gh/[project]",
    "status" : "success",
    "type" : "approval",
    "stopped_at" : "2024-03-06T09:33:58Z",
    "approval_request_id" : "a88020fd-c5c3-48ac-bc74-25a88593c08b"
  }, {
    "dependencies" : [ "a88020fd-c5c3-48ac-bc74-25a88593c08b" ],
    "job_number" : 127423,
    "id" : "e51528fe-b743-4231-b21c-3a54f225404e",
    "started_at" : "2024-03-06T09:35:09Z",
    "name" : "build_and_push_staging_docker_image-2",
    "project_slug" : "gh/[project]",
    "status" : "success",
    "type" : "build",
    "stopped_at" : "2024-03-06T09:41:10Z"
  }, {
    "dependencies" : [ "e51528fe-b743-4231-b21c-3a54f225404e" ],
    "job_number" : 127424,
    "id" : "f6c21172-105f-4c1b-a033-a631f859143d",
    "started_at" : "2024-03-06T09:41:12Z",
    "name" : "deploy_pr_staging",
    "project_slug" : "gh/[project]",
    "status" : "success",
    "type" : "build",
    "stopped_at" : "2024-03-06T09:42:23Z"
  } ]
}
```

- The job that responded with 404 was an [approval job](https://circleci.com/docs//workflows/#holding-a-workflow-for-a-manual-approval), which allows a user to manually approve the execution of an expensive workflow.
- The entire workflow seemed to be successful without any issues.

Although it is not written in the [Get job details API specification](https://circleci.com/docs/api/v2/index.html#operation/getJobDetails), it is not possible to view the details of the approval job even on the workflow web user interface, so it may not be possible to obtain the details of the approval job using the API.

**Comment by Kesin11** [comment]
@negipo 
I was able to reproduce same API response and fixed by #1125 
Also I just released fixed version [v6.0.3](https://github.com/Kesin11/CIAnalyzer/releases/tag/v6.0.3). Please try it.

**Comment by negipo** [comment]
After updating to v6.0.3 and running it, We were able to complete the race without any problems 😊
Thank you for your fix!
